packages/x-components/src/x-modules/queries-preview/components/query-preview.vue

```diff
@@ -223,12 +235,16 @@
         }
       );
 
-      const previewItemQuery = this.queryPreviewInfo.query;
+      const previewItemQuery = this.queryOfQueryPreview;
       const cachedQueryPreview = this.previewResults[previewItemQuery];
 
       // If the query has been saved it will emit load instead of the emitting the updated request.
-      if (cachedQueryPreview?.status === 'success' && this.persistInCache) {
-        this.$emit('load', this.queryPreviewInfo.query);
+      if (cachedQueryPreview?.status === 'success') {
+        this.$emit('load', previewItemQuery);
+        this.$x.emit('QueryPreviewMounted', this.queryOfQueryPreview, {
```

Comment:
Following the above comment, here we are using `previewItemQuery` and in the next line `this.queryOfQueryPreview` but both are the same. We should just use `this.queryOfQueryPreview` everywhere.